src/Microsoft.VisualStudio.Composition/RuntimeComposition.cs

```diff
@@ -498,7 +509,29 @@ public RuntimeExport(string contractName, TypeRef declaringTypeRef, MemberRef me
 
             public MemberRef MemberRef { get; private set; }
 
-            public TypeRef ExportedValueTypeRef { get; private set; }
+            public TypeRef ExportedValueTypeRef
+            {
+                get
+                {
+                    if (this.exportedValueTypeRef != null)
+                    {
+                        return this.exportedValueTypeRef;
+                    }
+                    else if (this.MemberRef == null)
+                    {
+                        return this.DeclaringTypeRef;
+                    }
+                    else
+                    {
+                        return TypeRef.Get(this.ExportedValueType, this.DeclaringTypeRef.Resolver);
```

Comment:
Similar here as an earlier comment: 
1. Should we cache this into the field for the next caller?
1. Can we invert this so that the TypeRef can be calculated without any assembly loads and the other non-ref property simply pulls on this property and resolves the result?